### DISCUSSION TITLE
watchdog enhancement?

### DISCUSSION THREAD

**Discussion by randyg503** [blakeblackshear/frigate#1254] [Ask A Question]
Could the watchdog routine be extended to watch both capture and detect style processes?  If I'm not mistaken, it seems like it currently watches only detect processes.  Thank you!

**Comment by blakeblackshear** [comment]
It definitely watches capture processes too. There are multiple watchdog threads because many parts of frigate run in separate dedicated processes.

**Reply by randyg503** [reply]
Oh sorry, I was not aware of that Blake, thank you for correcting me!  I am a bit baffled then why after manually/intentionally killing a capture process the camera stops updating in Frigate until i restart the container.   I commented in a somewhat related issue #1232 about my thoughts.

Thank you for Frigate and your excellent support!

**Reply by blakeblackshear** [reply]
How long did you wait? Do you see any errors in the logs? I'm surprised it doesn't react pretty quickly, but I should be able to reproduce this. Which roles were assigned to the ffmpeg process you killed manually?

**Reply by randyg503** [reply]
I've performed this test several times and waited various amounts of time ranging from 15 minutes to over night to see if the capture process would get respawned before giving up and restarting the container.  I don't see any errors in the log.  Roles assigned to the target PID for this test were clips, rtmp, and record on a 1920x1080 path and detect on a 352x240 path.  Please let me know if I can assist further.

Thank you for giving this simple test a shot.

**Reply by randyg503** [reply]
and just to be clear, I obtain the target PID for the kill command directly from the 'capture pid' column on the Debug panel

**Reply by blakeblackshear** [reply]
What command are you using to kill the process?

**Reply by randyg503** [reply]
so if the 'capture PID' was 203 for example, i would console into the running Frigate container (I use portainer) and issue 'kill 203' at the command prompt.
